lib/kube/proxy/forwarder.go

```diff
@@ -1207,7 +1207,7 @@ func (f *Forwarder) join(ctx *authContext, w http.ResponseWriter, req *http.Requ
 		client := &websocketClientStreams{uuid.New(), stream}
 		party := newParty(*ctx, stream.Mode, client)
 
-		err = session.join(party, true /* emitSessionJoinEvent */)
+		err = session.join(party)
```

Comment:
Removed this flag since `newUser != initiatingUser` is equivalent and used in the `join(party)` implementation.

Reply:
Oh I see. How about emitting the audit event and controls message when `len(s.parties) > 0`, i.e. the session has already been initiated? This logic can apply to both the SSH and Kube session management, making them a little more similar. I added commit b66ad27 to show this, but can revert and just use the `emitSessionJoinEvent` flag if you think this is digging a bigger hole.

Reply:
Chatting with @tigrato, he mentioned that SSH sessions can stay alive with the initiator leaving and joining (whereas Kube sessions cannot), which makes the above logic problematic. I'll revert https://github.com/gravitational/teleport/commit/b66ad2727098b5e310f24b9b7aee7860959162e5 as mentioned above, and use the original `emitSessionJoinEvent` flag.